examples/connext_dds/logging_config/c/logging_publisher.c

```diff
@@ -0,0 +1,232 @@
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* To customize publisher QoS, use
+    the configuration file USER_QOS_PROFILES.xml */
+    publisher = DDS_DomainParticipant_create_publisher(
+        participant, &DDS_PUBLISHER_QOS_DEFAULT, NULL /* listener */,
+        DDS_STATUS_MASK_NONE);
+    if (publisher == NULL) {
+        fprintf(stderr, "create_publisher error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* Register type before creating topic */
+    type_name = loggingTypeSupport_get_type_name();
+    retcode = loggingTypeSupport_register_type(
+        participant, type_name);
+    if (retcode != DDS_RETCODE_OK) {
+        fprintf(stderr, "register_type error %d\n", retcode);
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* To customize topic QoS, use
+    the configuration file USER_QOS_PROFILES.xml */
+    topic = DDS_DomainParticipant_create_topic(
+        participant, "Example logging",
+        type_name, &DDS_TOPIC_QOS_DEFAULT, NULL /* listener */,
+        DDS_STATUS_MASK_NONE);
+    if (topic == NULL) {
+        fprintf(stderr, "create_topic error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+    /* To customize data writer QoS, use
+    the configuration file USER_QOS_PROFILES.xml */
+    writer = DDS_Publisher_create_datawriter(
+        publisher, topic,
+        &DDS_DATAWRITER_QOS_DEFAULT, NULL /* listener */, DDS_STATUS_MASK_NONE);
+    if (writer == NULL) {
+        fprintf(stderr, "create_datawriter error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+    logging_writer = loggingDataWriter_narrow(writer);
+    if (logging_writer == NULL) {
+        fprintf(stderr, "DataWriter narrow error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* Create data sample for writing */
+    instance = loggingTypeSupport_create_data_ex(DDS_BOOLEAN_TRUE);
+    if (instance == NULL) {
+        fprintf(stderr, "loggingTypeSupport_create_data error\n");
+        publisher_shutdown(participant);
+        return -1;
+    }
+
+    /* Write data */
+    retcode = loggingDataWriter_write(
+        logging_writer, instance, &instance_handle);
+    if (retcode != DDS_RETCODE_OK) {
+        fprintf(stderr, "write error %d\n", retcode);
+    }
+
+    /*
+     * Force an warning writing:
+     * When using DDS_BY_SOURCE_TIMESTAMP_DESTINATIONORDER_QOS:
+     * If source timestamp is older than in previous write a warnings message
+     * will be logged.
+     */
+    retcode = loggingDataWriter_write_w_timestamp(
+        logging_writer, instance, &instance_handle, &sourceTimestamp);
+    if (retcode != DDS_RETCODE_BAD_PARAMETER) {
+        fprintf(stderr, "write error %d\n", retcode);
+    }
+
+    /* End - modifying the generated example to showcase the usage of
+     * the get_publishers API
```

Comment:
Is this comment correct? We are not showcasing the get_publishers API

Reply:
```
    /*
     * End - modifying the generated example to showcase the usage of
     * the Logging API.
     */
```